app/controllers/sign_up/personal_keys_controller.rb

```diff
@@ -3,23 +3,21 @@ class PersonalKeysController < ApplicationController
     include PersonalKeyConcern
 
     before_action :confirm_two_factor_authenticated
-    before_action :confirm_has_not_already_viewed_personal_key, only: [:show]
 
     def show
-      user_session.delete(:first_time_personal_key_view)
-      @code = create_new_code
+      @code = write_or_fetch_personal_key_in_session
```

Comment:
This change now opens up this endpoint to a CSRF attack. The benefit of the `before_action` we had before was to prevent the user's personal key from being changed if they already saw this view.

In order to allow the user to view the personal key page again, while also preventing CSRF attacks, we should generate the code beforehand and store it in the session, like we do in the other personal keys controller (`/users/personal_keys_controller.rb`), or perhaps we can get rid of this controller and use `UsersPersonalKeysController`.

Reply:
I like the idea of using the personal keys controller 🤔.

I'm going to flip this to WIP and explore some other options.